pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/impl/TransactionsBase.java

```diff
@@ -93,4 +105,109 @@ protected void internalGetCoordinatorStatus(AsyncResponse asyncResponse, boolean
                     "This Broker is not configured with transactionCoordinatorEnabled=true."));
         }
     }
+
+    protected void internalGetCoordinatorInternalStats(AsyncResponse asyncResponse, boolean authoritative,
+                                                       boolean metadata, int coordinatorId) {
+        try {
+            if (pulsar().getConfig().isTransactionCoordinatorEnabled()) {
+                TopicName topicName = TopicName.TRANSACTION_COORDINATOR_ASSIGN.getPartition(coordinatorId);
+                validateTopicOwnership(topicName, authoritative);
+                TransactionMetadataStore metadataStore = pulsar().getTransactionMetadataStoreService()
+                        .getStores().get(TransactionCoordinatorID.get(coordinatorId));
+                if (metadataStore instanceof MLTransactionMetadataStore) {
+                    asyncResponse.resume(getManageLedgerInternalStats(
+                            ((MLTransactionMetadataStore) metadataStore).getManagedLedger(),
+                            metadata, topicName.toString()).get());
+                } else {
+                    asyncResponse.resume(new RestException(METHOD_NOT_ALLOWED,
+                            "Broker don't use MLTransactionMetadataStore!"));
+                }
+            } else {
+                asyncResponse.resume(new RestException(SERVICE_UNAVAILABLE,
+                        "This Broker is not configured with transactionCoordinatorEnabled=true."));
+            }
+        } catch (Exception e) {
+            asyncResponse.resume(new RestException(e.getCause()));
+        }
+    }
+
+    private CompletableFuture<ManagedLedgerInternalStats> getManageLedgerInternalStats(ManagedLedger ledger,
+                                                                                       boolean includeLedgerMetadata,
+                                                                                       String topic) {
+        CompletableFuture<ManagedLedgerInternalStats> statFuture = new CompletableFuture<>();
+        ManagedLedgerInternalStats stats = new ManagedLedgerInternalStats();
+
+        ManagedLedgerImpl ml = (ManagedLedgerImpl) ledger;
+        stats.entriesAddedCounter = ml.getEntriesAddedCounter();
```

Comment:
what about moving this code to a method in ManagedLedgerImpl ?
or at least some utility method that is out of this TransactionsBase class.

this code belongs to ManagedLedgerImpl context, not the Transactions API